### PR TITLE
:seedling: Fix protoc build failures

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Fix protoc build failures by retries

Signed-off-by: naveensrinivasan <172697+naveensrinivasan@users.noreply.github.com>

#### What kind of change does this PR introduce?

(Is it a bug fix, feature, docs update, something else?)

- [ ] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**

- [ ] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note

```

### CODE REVIEW

**Comment by naveensrinivasan** [comment]
@azeemshaikh38 / @laurentsimon 👀

**Comment by naveensrinivasan** [comment]
FYI this is already used in other workflow https://github.com/ossf/scorecard/blob/7dcb3cb3e212a0c06f784e6f9da4d5d680e96bd3/.github/workflows/integration.yml#L78

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -104,7 +104,12 @@ jobs:
        with:
          go-version: '^1.17'
      - name: generate mocks
-       run: |
+       uses: nick-invision/retry@7f8f3d9f0f62fe5925341be21c2e8314fd4f7c7c
+       with:
+          max_attempts: 3
+          retry_on: error
+          timeout_minutes: 30
+          command: |
```
Comment (azeemshaikh38) [inline .github/workflows/main.yml:112]: Why do we need multiple retry? Shouldn't this only be needed for `build-proto`?

Comment (naveensrinivasan) [reply .github/workflows/main.yml:112]: These individual jobs have been failing, and that is the reason.

**Pull request merged**